src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -1032,7 +1035,10 @@ PosList ExpressionEvaluator::evaluate_expression_to_pos_list(const AbstractExpre
           });
         } break;
 
-        case PredicateCondition::Between:
+        case PredicateCondition::BetweenInclusive:
+        case PredicateCondition::BetweenLowerExclusive:
```

Comment:
(same here, you need to actually handle these four different cases, they can't just be routed to the exact same code path)

Reply:
should be fixed